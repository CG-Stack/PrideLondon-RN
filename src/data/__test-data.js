// @flow
// eslint-disable-next-line import/no-extraneous-dependencies
import { gen, sampleOne as sample } from "@rgbboy/testcheck";
import type { ValueGenerator } from "@rgbboy/testcheck";
import { DateTime } from "luxon";
import { FORMAT_CONTENTFUL_ISO, FORMAT_EUROPEAN_DATE } from "../lib/date";
import type { Event as EventDeprecated } from "./event-deprecated";
import type { Event, EventCategoryName } from "./event";
import type { FieldRef } from "./field-ref";
import type { HeaderBanner } from "./header-banner";
import type { ImageDetails } from "./image";
import type { Performance } from "./performance";
import type { Sponsor } from "./sponsor";
import { eventCategoryNames } from "./event";

type Options = {
  seed?: number
};

const defaultOptions: Options = {
  seed: 1
};

export const sampleOne = <A>(
  generator: ValueGenerator<A>,
  options: Options = defaultOptions
): A => {
  const seed = options.seed || defaultOptions.seed;
  return sample(generator, 30, seed);
};

export const sampleArrayOf = <A>(
  generator: ValueGenerator<A>
): (number => Array<A>) => (size: number) =>
  sample(gen.array(generator, { size }), 30, 1);

export const generateFieldRef: ValueGenerator<FieldRef> = gen({
  sys: gen({
    id: gen.alphaNumString
  })
});

const baseTime = 1530964800000; // July 7, 2018 12:00:00 PM GMT+00:00
const fiveMinutes = 300000;

export const generateDate: ValueGenerator<string> = gen.int.then(int =>
  DateTime.fromMillis(baseTime + int * int * int * fiveMinutes, {
    zone: "UTC"
  }).toFormat(FORMAT_EUROPEAN_DATE)
);

export const generateDateString: ValueGenerator<string> = gen.int.then(int =>
  DateTime.fromMillis(baseTime + int * int * int * fiveMinutes, {
    zone: "UTC"
  }).toFormat(FORMAT_CONTENTFUL_ISO)
);

export const generateImageURI: ValueGenerator<string> = gen.alphaNumString.then(
  name => `//red-badger.com/${name}.jpg`
);

// will change this when we refactor FieldRef
export const generateCMSFieldRef: ValueGenerator<mixed> = generateFieldRef;

export const generateImageDetails: ValueGenerator<ImageDetails> = gen({
  id: gen.alphaNumString,
  revision: 1,
  uri: generateImageURI.then(value => `https:${value}`),
  width: gen.intWithin(100, 1000),
  height: gen.intWithin(100, 1000)
});

export const generateCMSImage: ValueGenerator<mixed> = gen({
  sys: {
    id: gen.alphaNumString,
    type: "Asset",
    revision: 1
  },
  fields: {
    file: {
      "en-GB": {
        url: generateImageURI,
        details: {
          image: {
            height: gen.intWithin(100, 1000),
            width: gen.intWithin(100, 1000)
          }
        }
      }
    }
  }
});

export const generateHeaderBanner: ValueGenerator<HeaderBanner> = gen({
  contentType: "headerBanner",
  id: gen.alphaNumString,
  locale: "en-GB",
  revision: 1,
  fields: gen({
    heading: "heading",
    headingLine2: "headingLine2",
    subHeading: "subHeading",
    heroImage: generateFieldRef,
    backgroundColour: "#ff0000"
  })
});

export const generateCMSHeaderBanner: ValueGenerator<mixed> = gen({
  sys: {
    id: gen.alphaNumString,
    contentType: {
      sys: {
        id: "headerBanner"
      }
    },
    revision: 1
  },
  fields: {
    heading: {
      "en-GB": "heading"
    },
    headingLine2: {
      "en-GB": "headingLine2"
    },
    subHeading: {
      "en-GB": "subHeading"
    },
    heroImage: {
      "en-GB": generateFieldRef
    },
    backgroundColour: {
      "en-GB": "#ff0000"
    }
  }
});

export const generateEventCategory: ValueGenerator<
  EventCategoryName
> = gen.oneOf(eventCategoryNames);

export const generateEvent: ValueGenerator<Event> = gen({
  id: gen.alphaNumString,
  contentType: "event",
  locale: "en-GB",
  revision: 1,
  fields: gen({
    name: "name",
    eventCategories: gen.uniqueArray(generateEventCategory, {
      minSize: 1,
      maxSize: 5
    }),
    audience: gen.array(gen.alphaNumString, { minSize: 1, maxSize: 5 }),
    startTime: "2018-07-07T00:00+00:00",
    endTime: "2018-07-07T03:00+00:00",
    location: { lat: 0, lon: 10 },
    addressLine1: "addressLine1",
    addressLine2: "addressLine2",
    city: "city",
    postcode: "postcode",
    locationName: "locationName",
    eventPriceLow: 0,
    eventPriceHigh: 10,
    accessibilityOptions: gen.array(gen.alphaNumString, {
      minSize: 1,
      maxSize: 5
    }),
    eventDescription: "eventDescription",
    accessibilityDetails: "accessibilityDetails",
    email: "email",
    phone: "phone",
    ticketingUrl: "ticketingUrl",
    venueDetails: gen.array(gen.alphaNumString, { minSize: 1, maxSize: 5 }),
    individualEventPicture: generateFieldRef,
    eventsListPicture: generateFieldRef,
    performances: gen.array(generateFieldRef, { minSize: 1, maxSize: 5 }),
    recurrenceDates: gen.array(generateDate, { minSize: 1, maxSize: 5 })
  })
});

export const generateCMSEvent: ValueGenerator<EventDeprecated> = gen({
  sys: gen({
    id: gen.alphaNumString,
    contentType: {
      sys: {
        id: "event"
      }
    },
    revision: 1
  }),
  fields: gen({
    name: { "en-GB": "name" },
    eventCategories: {
      "en-GB": gen.uniqueArray(generateEventCategory, {
        minSize: 1,
        maxSize: 5
      })
    },
    audience: {
      "en-GB": gen.array(gen.alphaNumString, { minSize: 1, maxSize: 5 })
    },
    startTime: { "en-GB": "2018-07-07T00:00+00:00" },
    endTime: { "en-GB": "2018-07-07T03:00+00:00" },
    location: { "en-GB": { lat: 0, lon: 10 } },
    addressLine1: { "en-GB": "addressLine1" },
    addressLine2: { "en-GB": "addressLine2" },
    city: { "en-GB": "city" },
    postcode: { "en-GB": "postcode" },
    locationName: { "en-GB": "locationName" },
    eventPriceLow: { "en-GB": 0 },
    eventPriceHigh: { "en-GB": 10 },
    accessibilityOptions: {
      "en-GB": gen.array(gen.alphaNumString, { minSize: 1, maxSize: 5 })
    },
    eventDescription: { "en-GB": "eventDescription" },
    accessibilityDetails: { "en-GB": "accessibilityDetails" },
    email: { "en-GB": "email" },
    phone: { "en-GB": "phone" },
    ticketingUrl: { "en-GB": "ticketingUrl" },
    venueDetails: {
      "en-GB": gen.array(gen.alphaNumString, { minSize: 1, maxSize: 5 })
    },
    individualEventPicture: gen({ "en-GB": generateFieldRef }),
    eventsListPicture: gen({ "en-GB": generateFieldRef }),
    performances: {
      "en-GB": gen.array(generateFieldRef, { minSize: 1, maxSize: 5 })
    },
    recurrenceDates: {
      "en-GB": gen.array(generateDate, { minSize: 1, maxSize: 5 })
    }
  })
});

export const generateEventMinimum: ValueGenerator<Event> = gen({
  id: gen.alphaNumString,
  contentType: "event",
  locale: "en-GB",
  revision: 1,
  fields: gen({
    name: "name",
    eventCategories: gen.uniqueArray(generateEventCategory, {
      minSize: 1,
      maxSize: 5
    }),
    audience: [],
    startTime: "2018-07-07T00:00+00:00",
    endTime: "2018-07-07T03:00+00:00",
    location: { lat: 0, lon: 10 },
    addressLine1: gen.null,
    addressLine2: gen.null,
    city: gen.null,
    postcode: gen.null,
    locationName: "locationName",
    eventPriceLow: 0,
    eventPriceHigh: 10,
    accessibilityOptions: [],
    eventDescription: "eventDescription",
    accessibilityDetails: gen.null,
    email: gen.null,
    phone: gen.null,
    ticketingUrl: gen.null,
    venueDetails: [],
    individualEventPicture: generateFieldRef,
    eventsListPicture: generateFieldRef,
    performances: [],
    recurrenceDates: []
  })
});

export const generateCMSEventMinimum: ValueGenerator<mixed> = gen({
  sys: gen({
    id: gen.alphaNumString,
    contentType: {
      sys: {
        id: "event"
      }
    },
    revision: 1
  }),
  fields: gen({
    name: { "en-GB": "name" },
    eventCategories: {
      "en-GB": ["Cabaret and Variety", "Music"]
    },
    startTime: { "en-GB": "2018-07-07T00:00+00:00" },
    endTime: { "en-GB": "2018-07-07T03:00+00:00" },
    location: { "en-GB": { lat: 0, lon: 10 } },
    locationName: { "en-GB": "locationName" },
    eventPriceLow: { "en-GB": 0 },
    eventPriceHigh: { "en-GB": 10 },
    eventDescription: { "en-GB": "eventDescription" },
    individualEventPicture: gen({ "en-GB": generateFieldRef }),
    eventsListPicture: gen({ "en-GB": generateFieldRef })
  })
});

export const generatePerformance: ValueGenerator<Performance> = gen({
  contentType: "performance",
  id: gen.alphaNumString,
  locale: "en-GB",
  revision: 1,
  fields: gen({
    title: "title",
    startTime: generateDateString
  })
});

export const generateCMSPerformance: ValueGenerator<mixed> = gen({
  sys: {
    id: gen.alphaNumString,
    contentType: {
      sys: {
        id: "performance"
      }
    },
    revision: 1
  },
  fields: gen({
    title: {
      "en-GB": "title"
    },
    startTime: gen({
      "en-GB": generateDateString
    })
  })
});

export const generateSponsor: ValueGenerator<Sponsor> = gen({
  contentType: "sponsor",
  id: gen.alphaNumString,
  locale: "en-GB",
  revision: 1,
  fields: gen({
    sponsorName: "sponsorName",
    sponsorLogo: generateFieldRef,
    sponsorUrl: "sponsorUrl",
    sponsorLevel: "Headline"
  })
});

export const generateCMSSponsor: ValueGenerator<mixed> = gen({
  sys: {
    id: gen.alphaNumString,
    contentType: {
      sys: {
        id: "sponsor"
      }
    },
    revision: 1
  },
  fields: {
    sponsorName: { "en-GB": "sponsorName" },
    sponsorLogo: { "en-GB": generateFieldRef },
    sponsorUrl: { "en-GB": "sponsorUrl" },
    sponsorLevel: { "en-GB": "Headline" }
  }
});
