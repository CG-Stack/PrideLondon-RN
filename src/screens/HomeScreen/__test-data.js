// @flow
import type { Event } from "../../data/event";
import type { HeaderBanner } from "../../data/header-banner";

export const generateHeaderBanners = (count: number): HeaderBanner[] =>
  Array.from(Array(count)).map(
    (_, i) =>
      ({
        sys: {
          id: String(i + 1)
        },
        fields: {
          heading: {
            "en-GB": "Pride"
          },
          headingLine2: {
            "en-GB": "Festival"
          },
          subHeading: {
            "en-GB": "from - to"
          },
          heroImage: {
            "en-GB": {
              sys: {
                id: `asset${i + 1}`
              }
            }
          },
          backgroundColour: {
            "en-GB": "#ff0000"
          }
        }
      }: any)
  );

export const generateEvents = (count: number): Event[] =>
  Array.from(Array(count)).map(
    (_, i) =>
      ({
        sys: {
          id: String(i + 1)
        },
        fields: {
          name: {
            "en-GB": "some other"
          },
          eventsListPicture: {
            "en-GB": {
              sys: {
                id: `asset${i + 1}`
              }
            }
          },
          startTime: {
            "en-GB": "2018-07-10T00:00"
          },
          eventCategories: {
            "en-GB": ["Cabaret & Variety", "Music"]
          }
        }
      }: any)
  );
