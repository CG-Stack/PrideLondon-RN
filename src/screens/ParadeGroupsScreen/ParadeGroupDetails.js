// @flow
import React, { PureComponent } from "react";
import type { ElementRef } from "react";
import { Animated, Easing, Linking, StyleSheet, View } from "react-native";
import type { ParadeGroup } from "../../data/parade-group";
import Text from "../../components/Text";
import TextLink from "../../components/TextLink";
import Touchable from "../../components/Touchable";
import text from "../../constants/text";
import chevron from "../../../assets/images/chevronRightGreen.png";

type Props = {
  paradeGroup: ParadeGroup
};

type State = {
  animation: Object,
  collapsed: boolean,
  contentHeight: number
};

const renderURL = (label, url) => (
  <Touchable onPress={() => Linking.openURL(url)}>
    <TextLink>{label}</TextLink>
  </Touchable>
);

const spinInterpolation = {
  inputRange: [0, 1],
  outputRange: ["90deg", "-90deg"]
};

class ParadeGroupDetails extends PureComponent<Props, State> {
  constructor() {
    super();
    // $FlowFixMe: For some reason flow doesn't know about React.createRef.
    this.collapsable = React.createRef();
  }

  state = {
    animation: new Animated.Value(0),
    collapsed: true,
    contentHeight: 0
  };

  componentDidUpdate(prevProps: Props, prevState: State) {
    if (prevState.collapsed === this.state.collapsed) return;

    const toValue = this.state.collapsed ? 0 : 1;
    const { animation } = this.state;

    Animated.timing(animation, {
      toValue,
      duration: 200,
      easing: Easing.ease
    }).start();
  }

  collapsable: ElementRef<typeof View>;

  toggleCollapse = () => {
    if (this.collapsable.current && this.state.collapsed) {
      this.collapsable.current.measure((a, b, width, height) => {
        this.setState({
          collapsed: !this.state.collapsed,
          contentHeight: height
        });
      });
    } else {
      this.setState({
        collapsed: !this.state.collapsed
      });
    }
  };

  render() {
    const {
      name,
      section,
      facebookUrl,
      twitterUrl,
      websiteUrl
    } = this.props.paradeGroup.fields;
    const { animation, collapsed, contentHeight } = this.state;
    const spin = animation.interpolate(spinInterpolation);
    const height = animation.interpolate({
      inputRange: [0, 1],
      outputRange: [4, contentHeight] // must have some height when collapsed to be able to measure correctly.
    });
    return (
      <View style={styles.container}>
        <Touchable
          accessibilityLabel={
            collapsed
              ? text.paradeGroups.expandAccessibilityLabel
              : text.paradeGroups.collapseAccessibilityLabel
          }
          onPress={this.toggleCollapse}
          style={styles.collapseButton}
        >
          <Animated.Image
            style={[styles.chevron, { transform: [{ rotate: spin }] }]}
            source={chevron}
          />
        </Touchable>
        <View style={styles.details}>
          <Text type="h4">{name}</Text>
          <Animated.View style={[styles.animatedContainer, { height }]}>
            <View style={styles.collapsable} ref={this.collapsable}>
              <Text style={styles.section}>{section}</Text>
              {facebookUrl &&
                renderURL(text.paradeGroups.linkLabelFacebook, facebookUrl)}
              {twitterUrl &&
                renderURL(text.paradeGroups.linkLabelTwitter, twitterUrl)}
              {websiteUrl &&
                renderURL(text.paradeGroups.linkLabelWebsite, websiteUrl)}
            </View>
          </Animated.View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0,
    flexGrow: 1,
    flexShrink: 1,
    flexDirection: "row",
    alignItems: "flex-start"
  },
  chevron: {
    width: 24,
    height: 24
  },
  collapseButton: {},
  details: {
    flex: 0,
    flexGrow: 1,
    flexShrink: 1,
    flexDirection: "column",
    paddingTop: 10,
    paddingBottom: 10
  },
  animatedContainer: {
    overflow: "hidden"
  },
  collapsable: {
    minHeight: 32 // required to stop text from collapsing
  },
  section: {
    paddingTop: 8
  }
});

export default ParadeGroupDetails;
