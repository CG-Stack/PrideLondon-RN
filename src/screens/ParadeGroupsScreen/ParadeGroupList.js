// @flow
import React, { Component } from "react";
import { StyleSheet, SectionList, View } from "react-native";
import type { SectionBase } from "react-native/Libraries/Lists/SectionList";
import type { ParadeGroup } from "./parade-group";
import ContentPadding from "../../components/ContentPadding";
import SectionHeader from "../../components/SectionHeader";
import Text from "../../components/Text";
import { whiteColor } from "../../constants/colors";

type Props = {
  paradeGroups: ParadeGroup[][],
  testID?: string
};

type Section = SectionBase<ParadeGroup> & { index: number };

type RenderItemInfo = {
  item: ParadeGroup, // eslint-disable-line react/no-unused-prop-types
  index: number,
  section: Section
};

type RenderSectionInfo = {
  section: Section // eslint-disable-line react/no-unused-prop-types
};

const sectionTitle = (paradeGroup: ParadeGroup): string =>
  paradeGroup.fields.name.charAt(0).toUpperCase();

const sections = (sectionedParadeGroups: ParadeGroup[][]): Section[] =>
  sectionedParadeGroups.map((paradeGroups, index) => ({
    data: paradeGroups,
    key: sectionTitle(paradeGroups[0]),
    index
  }));

const getId = (item: { id: string }): string => item.id;

class ParadeGroupList extends Component<Props> {
  static defaultProps = {
    testID: undefined
  };

  shouldComponentUpdate(nextProps: Props) {
    const { paradeGroups } = this.props;
    const { paradeGroups: nextParadeGroups } = nextProps;

    return paradeGroups !== nextParadeGroups;
  }

  itemSeparator = () => <View style={styles.itemSeparator} />;

  sectionSeparator = () => <View style={styles.sectionSeparator} />;

  keyExtractor = getId;

  renderItem = ({ item }: RenderItemInfo) => (
    <ContentPadding>
      <Text>Name: {item.fields.name}</Text>
    </ContentPadding>
  );

  renderSectionHeader = ({ section }: RenderSectionInfo) => (
    <SectionHeader title={sectionTitle(section.data[0])} />
  );

  renderSectionFooter = () => <View style={styles.sectionFooter} />;

  render() {
    const { paradeGroups, testID } = this.props;

    return (
      <SectionList
        stickySectionHeadersEnabled
        sections={sections(paradeGroups)}
        renderSectionHeader={this.renderSectionHeader}
        renderSectionFooter={this.renderSectionFooter}
        renderItem={this.renderItem}
        keyExtractor={this.keyExtractor}
        contentContainerStyle={styles.container}
        ItemSeparatorComponent={this.itemSeparator}
        SectionSeparatorComponent={this.sectionSeparator}
        testID={testID}
        windowSize={10}
      />
    );
  }
}

const styles = StyleSheet.create({
  itemSeparator: {
    height: 12
  },
  sectionSeparator: {
    height: 6
  },
  container: {
    paddingTop: 0,
    backgroundColor: whiteColor
  },
  sectionFooter: {
    height: 6
  }
});

export default ParadeGroupList;
