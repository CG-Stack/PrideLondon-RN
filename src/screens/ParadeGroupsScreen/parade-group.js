// @flow
import type { Maybe } from "../../lib/maybe";

export type ParadeGroup = {
  // important to keep this at the top level so type refinement works
  contentType: "paradeGroup",
  id: string,
  locale: string,
  revision: number,
  fields: {
    name: string,
    section: string,
    facebookUrl: Maybe<string>,
    twitterUrl: Maybe<string>,
    websiteUrl: Maybe<string>
  }
};
