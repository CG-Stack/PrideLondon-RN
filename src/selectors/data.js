// @flow
import type { Event, Events } from "../data/event";
import type { Performance, Performances } from "../data/performance";
import type { State as DataState } from "../reducers/data";

type ObjectWithId<A> = {
  id: string
} & A;

const reduceToMapHelp = <A>(
  acc: { [id: string]: ObjectWithId<A> },
  item: ObjectWithId<A>
): { [id: string]: ObjectWithId<A> } => {
  acc[item.id] = item; // intentional mutation as this happens in a reduce
  return acc;
};

export const selectEvents = (data: DataState): Array<Event> => data.events;

export const selectEventsMap = (events: Array<Event>): Events =>
  events.reduce(reduceToMapHelp, {});

export const selectEventById = (events: Events, id: string): ?Event =>
  events[id];

export const selectPerformancesMap = (data: DataState): Performances =>
  data.performances;

export const selectPerformanceById = (
  performances: Performances,
  id: string
): ?Performance => performances[id];
