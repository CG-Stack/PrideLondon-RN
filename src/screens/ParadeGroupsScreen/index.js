// @flow
import { connect } from "react-redux";
import type { Connector } from "react-redux";
import type { ParadeGroup } from "../../data/parade-group";
import type { State } from "../../reducers";
import withIsFocused from "../../components/WithIsFocused";
import Component from "./component";

import { generateParadeGroup, sampleArrayOf } from "../../data/__test-data";

type OwnProps = {
  isFocused: boolean
};

type Props = {
  paradeGroups: ParadeGroup[]
};

let cache: Props;

// @TODO: REMOVE THIS BEFORE MERGING TO MASTER
const paradeGroups = sampleArrayOf(generateParadeGroup)(200);

paradeGroups[0].fields.name = "Above the Stag Theatre";

// Note we must add a return type here for react-redux connect to work
// with flow correctly. If not provided is silently fails if types do
// not line up. See https://github.com/facebook/flow/issues/5343
const mapStateToProps = (state: State, { isFocused }: OwnProps): Props => {
  if (!cache || isFocused) {
    cache = {
      paradeGroups
    };
  }
  return cache;
};

const connector: Connector<OwnProps, Props> = connect(mapStateToProps);

export const Container = connector(Component);

export default withIsFocused(Container);
