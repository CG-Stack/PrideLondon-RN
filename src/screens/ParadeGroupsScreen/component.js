// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import ParadeGroupList from "./ParadeGroupList";
import type { ParadeGroup } from "./parade-group";
import { bgColor } from "../../constants/colors";

export const groupByName = (paradeGroups: ParadeGroup[]): ParadeGroup[][] => [
  paradeGroups
];

type Props = {
  paradeGroups: ParadeGroup[]
};

const ParadeGroupsScreen = (props: Props) => (
  <View style={styles.container} testID="parade-groups-screen">
    <ParadeGroupList paradeGroups={groupByName(props.paradeGroups)} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: bgColor
  }
});

export default ParadeGroupsScreen;
