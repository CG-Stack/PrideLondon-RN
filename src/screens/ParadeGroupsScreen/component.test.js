// @flow
import React from "react";
import { shallow } from "enzyme";
import ParadeGroupsScreen from "./component";
import { generateParadeGroup, sampleOne } from "../../data/__test-data";

it("renders correctly", () => {
  const paradeGroups = [
    sampleOne(generateParadeGroup, { seed: 5432 }),
    sampleOne(generateParadeGroup, { seed: 4534 }),
    sampleOne(generateParadeGroup, { seed: 3456 })
  ];
  const output = shallow(<ParadeGroupsScreen paradeGroups={paradeGroups} />);
  expect(output).toMatchSnapshot();
});
