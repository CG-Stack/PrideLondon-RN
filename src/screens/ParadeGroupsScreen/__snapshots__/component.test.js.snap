// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="parade-groups-screen"
>
  <ParadeGroupList
    paradeGroups={
      Array [
        Array [
          Object {
            "contentType": "paradeGroup",
            "fields": Object {
              "facebookUrl": "https://red-badger.com/1r59u1Oj4T7D55BUw",
              "name": "1x2yj",
              "section": "Section F",
              "twitterUrl": "https://red-badger.com/9bP1RB90H",
              "websiteUrl": "https://red-badger.com/eip8IG6Kik565CkXG",
            },
            "id": "d18Cp545J4C996o5",
            "locale": "en-GB",
            "revision": 1,
          },
          Object {
            "contentType": "paradeGroup",
            "fields": Object {
              "facebookUrl": null,
              "name": "99b3j372i",
              "section": "Section F",
              "twitterUrl": "https://red-badger.com/3K6989xRF5UXeYRQ43y4",
              "websiteUrl": "https://red-badger.com/U8vZ3SjWR8f763prj6la40kMR",
            },
            "id": "642gdEs5xmkdO5v541Oau11NM7T",
            "locale": "en-GB",
            "revision": 1,
          },
          Object {
            "contentType": "paradeGroup",
            "fields": Object {
              "facebookUrl": null,
              "name": "w6PEBGjV01WHhj9",
              "section": "Section A",
              "twitterUrl": null,
              "websiteUrl": null,
            },
            "id": "LCUP4D33RDtZPmk2q9b6yvSStOJkzA",
            "locale": "en-GB",
            "revision": 1,
          },
        ],
      ]
    }
  />
</Component>
`;
